### PR TITLE
Simplify API

### PR DESCRIPTION
**Pull request by lukasz-walkiewicz**

### CODE REVIEW

File: neptune/__init__.py [hunk 1]
```diff
@@ -13,3 +13,87 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from neptune import projects, experiments
+from neptune.internal.structs.stack import Stack
+from neptune.sessions import Session
+
+session = None
```
Comment (pitercl) [inline neptune/__init__.py]: Locks, please.

File: neptune/experiments.py [hunk 2]
```diff
@@ -40,7 +41,7 @@ class Experiment(object):
         Instantiate a session.
 
         >>> from neptune.session import Session
-        >>> session = Session()
+        >>> current_session = Session()
```
Comment (pitercl) [inline neptune/experiments.py]: Why make it more complex?

File: neptune/experiments.py [hunk 3]
```diff
@@ -439,3 +442,26 @@ def _create_channel(self, channel_name, channel_type):
         channel = self._client.create_channel(self.internal_id, channel_name, channel_type)
         self._leaderboard_entry.add_channel(channel)
         return channel
+
+
+current_experiment = None
+
+_run_stack = Stack()
```
Comment (pitercl) [inline neptune/experiments.py]: `_experiment_stack` maybe?

File: neptune/internal/structs/stack.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+class Stack(object):
```
Comment (pitercl) [inline neptune/internal/structs/stack.py]: What's wrong with using a list? Why do we need this wrapper?

File: neptune/experiments.py [hunk 5]
```diff
@@ -439,3 +442,26 @@ def _create_channel(self, channel_name, channel_type):
...
+
+
+def push_new_experiment(new_experiment):
```
Comment (pitercl) [inline neptune/experiments.py]: Locks, please.

File: neptune/projects.py [hunk 6]
```diff
@@ -267,7 +267,7 @@ def get_experiment_groups(self):
 
     # pylint:disable=unused-argument
     def create_experiment(self,
-                          name,
+                          name="Untitled",
```
Comment (pitercl) [inline neptune/projects.py]: It seems that this default name should be in a single place.

File: neptune/__init__.py [hunk 7]
```diff
@@ -13,3 +13,87 @@
...
+                      properties=None,
+                      tags=None,
+                      upload_source_files=None,
+                      abort_callback=None,
+                      send_hardware_metrics=True,
+                      run_monitoring_thread=True,
+                      handle_uncaught_exceptions=True):
+    # pylint: disable=global-statement
+    global project
+    return experiments.push_new_experiment(
```
Comment (pitercl) [inline neptune/__init__.py]: `push_new_experiment` is also called in `project.create_exp`.

File: neptune/sessions.py [hunk 8]
```diff
@@ -64,6 +65,15 @@ def get_project(self, project_qualified_name):
             name=project.name
         )
 
+    def get_default_project(self, organization_name=None):
```
Comment (pitercl) [inline neptune/sessions.py]: Why do we need this in `Session`'s API? I'd drop it.

File: neptune/__init__.py (hunk 1 above)
Comment (pitercl) [reply neptune/__init__.py]: I know that assignment is atomic, but you don't just assign:

```
if session is None:
        session = init(project_qualified_name=project_qualified_name)
```

File: neptune/sessions.py (hunk 8 above)
Comment (lukasz-walkiewicz) [reply neptune/sessions.py]: `get_default_project` is called when user initializes session without `project_qualified_name` argument. I can drop it but it that case we'll need to find another solution to support flow like:
```python
import neptune

neptune.init()
neptune.create_experiment(...)
```

File: neptune/internal/structs/stack.py (hunk 4 above)
Comment (lukasz-walkiewicz) [reply neptune/internal/structs/stack.py]: IMO it looks better this way but I can use plain list if you want me to.

Comment (pitercl) [reply neptune/internal/structs/stack.py]: I'd prefer less code than more.

File: neptune/sessions.py (hunk 8 above)
Comment (pitercl) [reply neptune/sessions.py]: I think it's better to inline it in `init()`. We don't want to expose such functionality, it's here only to facilitate the simplest-possible API.

**Pull request merged**